keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,612 @@
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Disabling the endpoints controllers in a cluster where some
+third-party components depend on them could have arbitrarily bad
+effects.
+
+###### What specific metrics should inform a rollback?
+
+We may add metrics monitoring usage of the v1.Endpoints API, but
+ideally you would have looked at those metrics _before_ disabling the
+Endpoints controller.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+N/A
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The feature would be enabled by the operator.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details: 
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+There are no specific SLOs, but kube-controller-manager,
+kube-apiserver, and etcd should use less CPU, and etcd should use less
+disk space, if the controllers are disabled.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+(See above about metrics informing a rollback.)
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No; the goal of this KEP is to drastically reduce the overall size of
+the API database.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No; the goal of this KEP is to drastically reduce the overall size of
+the API database, and to somewhat reduce the CPU usage of
+kube-controller-manager, kube-apiserver, and etcd.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A
+
+###### What are other known failure modes?
+
+None
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+- Initial proposal: 2024-11-21
+
```

Comment:
You are missing Drawbacks, Alternatives, and Infrastrucuture needed (Optional).

Reply:
There is really nothing interesting to say (and "Infrastructure needed (Optional)" is optional, and explicitly intended to be deleted if not used)